eng/tools/lint-diff/test/generateReport.test.ts

```diff
@@ -1,17 +1,44 @@
-import { test, describe, expect, vi, afterEach } from "vitest";
+import { beforeEach, test, describe, expect, vi } from "vitest";
 import {
-  iconFor,
-  getLine,
-  getFile,
+  compareLintDiffViolations,
+  generateAutoRestErrorReport,
+  generateLintDiffReport,
   getDocUrl,
+  getFile,
   getFileLink,
-  generateReport,
+  getLine,
   getPathSegment,
-  compareLintDiffViolations,
+  iconFor,
 } from "../src/generateReport.js";
-import { Source, LintDiffViolation, BeforeAfter, AutorestRunResult } from "../src/lintdiff-types.js";
+import {
+  Source,
+  LintDiffViolation,
+  BeforeAfter,
+  AutorestRunResult,
+  AutoRestMessage,
+} from "../src/lintdiff-types.js";
 import { isWindows } from "./test-util.js";
 
+import { vol } from "memfs";
+
+vi.mock("node:fs/promises", async () => {
+  const memfs = await vi.importActual("memfs") as typeof import("memfs");
+  return {
+    ...memfs.fs.promises,
+  };
+});
+
+import { readFile } from "fs/promises";
+
+vi.mock("../src/util.js", async () => {
+  const original = await vi.importActual("../src/util.js");
+  return {
+    ...original,
+    getDependencyVersion: vi.fn().mockResolvedValue("1.0.0"),
+    getPathToDependency: vi.fn().mockResolvedValue("path/to/dependency"),
+  };
+});
```

Comment:
Mocking `getPathToDependency` which uses `import.meta.resolve` (not supported in vitest) and `getDependencyVersion` which requires a path that has a parent `package.json`. 

The outputs of these functions are not important to the tests in this case, but `getPathToDependency` will throw because it uses `import.meta.resolve`.